server/build/webpack/base.config.js

```diff
@@ -0,0 +1,318 @@
+        fallback: {
+          loader: 'style-loader',
+          options: {
+            sourceMap: true,
+            importLoaders: 1
+          }
+        }
+      }) : [])
+    ].filter(Boolean)
+  }
+
+  const babelLoaderOptions = babelConfig(dir, isServer)
+
+  let webpackConfig = {
+    devtool: dev ? 'cheap-module-source-map' : false,
+    // devtool: 'source-map',
+    name: isServer ? 'server' : 'client',
+    cache: true,
+    profile: true,
+    target: isServer ? 'node' : 'web',
+    externals: externalsConfig(dir, isServer),
+    context: dir,
+    entry: async () => {
+      const pages = await getPages(dir, {dev, isServer})
+      const mainJS = require.resolve(`../../../client/next${dev ? '-dev' : ''}`)
+      const clientConfig = !isServer ? {
+        'main.js': [
+          dev && !isServer && path.join(__dirname, '..', '..', '..', 'client', 'webpack-hot-middleware-client'),
+          dev && !isServer && path.join(__dirname, '..', '..', '..', 'client', 'on-demand-entries-client'),
+          mainJS
+        ].filter(Boolean)
+      } : {}
+      return {
+        ...clientConfig,
+        ...pages
+      }
+    },
+    output: {
+      path: path.join(dir, '.next', isServer ? 'dist' : ''),
+      filename: '[name]',
+      libraryTarget: 'commonjs2',
+      publicPath: `/_next/webpack/`,
+      // This saves chunks with the name given via require.ensure()
+      chunkFilename: '[name]-[chunkhash].js',
+      sourceMapFilename: '[file].map?[contenthash]'
+    },
+    performance: { hints: false },
+    resolve: {
+      extensions: ['.js', '.jsx', '.json', '.ts', '.tsx'],
+      modules: [
+        nextNodeModulesDir,
+        'node_modules'
+      ],
+      alias: {
+        'next': nextDir,
+        'react-dom': dev ? 'react-dom/cjs/react-dom.development.js' : 'react-dom/cjs/react-dom.production.min.js'
+      }
+    },
+    resolveLoader: {
+      modules: [
+        nextNodeModulesDir,
+        'node_modules',
+        path.join(__dirname, '..', 'loaders')
+      ]
+    },
+    module: {
+      rules: [
+        dev && !isServer && {
+          test: /\.(js|jsx)(\?[^?]*)?$/,
+          loader: 'hot-self-accept-loader',
+          include: [
+            path.join(dir, 'pages'),
+            nextPagesDir
+          ]
+        },
+        {
+          test: /\.+(js|jsx)$/,
+          include: [dir],
+          exclude: /node_modules/,
+          use: {
+            loader: 'babel-loader',
+            options: babelLoaderOptions
+          }
+        },
+        {
+          test: /\.+(ts|tsx)$/,
+          include: [dir],
+          exclude: /node_modules/,
+          use: [
+            {
+              loader: 'babel-loader',
+              options: babelLoaderOptions
+            },
+            {
+              loader: 'ts-loader',
+              options: {
+                transpileOnly: true
+              }
+            }
+          ]
+        },
+        {
+          test: /\.css$/,
```

Comment:
Shall we move CSS support as I mentioned above?
Into another package(which used in our webpack config) or as an example.